app/src/main/kotlin/com/hippo/ehviewer/ui/screen/SearchBarScreen.kt

```diff
@@ -244,9 +246,15 @@ fun SearchBarScreen(
                     },
                     trailingIcon = {
                         if (expanded) {
-                            if (searchFieldState.text.isNotEmpty()) {
-                                IconButton(onClick = { searchFieldState.clearText() }) {
-                                    Icon(Icons.Default.Close, contentDescription = null)
+                            AnimatedContent(targetState = searchFieldState.text.isNotEmpty()) { hasText ->
+                                if (hasText) {
+                                    IconButton(onClick = { searchFieldState.clearText() }) {
+                                        Icon(Icons.Default.Close, contentDescription = null)
+                                    }
+                                } else {
+                                    IconButton(onClick = { navigate(ImageSearchScreenDestination) }) {
+                                        Icon(Icons.Default.ImageSearch, contentDescription = null)
+                                    }
```

Comment:
Should we always show image search button or only show it when the search text is empty?

Reply:
I suggest we show only when search text is empty as we will eventually convert it to dialog, should not imply "the image search will take textfield info"